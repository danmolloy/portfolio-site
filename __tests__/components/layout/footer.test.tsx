import Footer from "@/components/layout/footer";
import { act, fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom"
import { animateScroll } from 'react-scroll'

jest.mock("react-scroll", () => ({
  animateScroll: {
    scrollToTop: jest.fn()
  }
}))

describe("<Footer />", () => {
  beforeEach(() => {
    render(<Footer />)
  })
  it("footer is in the document", () => {
    const footer = screen.getByTestId("footer")
    expect(footer).toBeInTheDocument()
  })
  it("back to top btn is in the document and calls scrollToTop on click", () => {
    const backToTopBtn = screen.getByText("Back to top")
    expect(backToTopBtn).toBeInTheDocument()
    act(() => {
      fireEvent.click(backToTopBtn)
    })
    expect(animateScroll.scrollToTop).toHaveBeenCalledWith({
      smooth: true, 
      duration: 750
    })
  })
  it("design and building credit is in the document", () => {
    const buildCredit = screen.getByText("Designed and built by Daniel Molloy")
    expect(buildCredit).toBeInTheDocument()
  })
  it("matches snapshot", () => {
    const footer = screen.getByTestId("footer")
    expect(footer).toMatchSnapshot()
  })
})