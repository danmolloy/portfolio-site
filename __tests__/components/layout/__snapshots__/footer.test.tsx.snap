// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<div
  class="w-full h-20 dark:bg-zinc-900 flex sm:flex-row flex-col sm:justify-between p-2 items-center border-t"
  data-testid="footer"
>
  <button
    class="hover:underline font-display text-sm font-light hover:cursor-pointer active:text-zinc-400 m-1 "
  >
    Back to top
  </button>
  <p
    class="text-sm text-zinc-400 font-display"
  >
    Designed and built by Daniel Molloy
  </p>
</div>
`;
